src/modm/platform/dma/stm32/module.lb

```diff
@@ -16,15 +17,45 @@ def init(module):
     module.name = ":platform:dma"
     module.description = "Direct Memory Access (DMA)"
 
+
 def prepare(module, options):
     device = options[":target"]
-    if not device.has_driver("dma:stm32*"):
-        return False
 
-    # FIXME the driver is for L4 only
-    if device.identifier["family"] not in ["l4"]:
+    if (not device.has_driver("dma")) or device.get_driver("dma")["type"] not in \
+            [
+            "stm32-channel-request",
+            "stm32-channel",
+    ]:
         return False
-    if device.identifier["name"] not in ["12", "22", "31", "32", "33", "42", "43", "51", "52", "62", "75", "76", "86", "96", "A6"]:
+
+    did = device.identifier
+
+    # Enable DMA for all but some devices...
+    if not (
+        (did["family"] in ["f0"]
+         # STM32F09x has shared interrupts between DMA1 and DMA2, not supported...
+         and did["name"] not in ["30", "71", "72", "78", "91", "98"])
+        or (did["family"] in ["f1"] and
+            did["name"] in ["02", "05", "07"]
+            or
+            (did["name"] in ["00", "01", "03"]
+             and did["pin"] in ["r", "v", "z"]
+             and did["size"] in ["c", "d", "e", "f", "g"]
+             )
+            )
+        or (did["family"] in ["f3"] and
+            did["name"] not in ["73", "78"]
+            and
+            (did["name"] in ["02"]
+             and did["pin"] not in ["c"]
+             and did["size"] not in ["b", "c"]
+             )
+            )
+        or (did["family"] in ["l0"])
+        or (did["family"] in ["l1"]
+            and did["size"] not in ["c"])
```

Comment:
The _stm32l100r**c**t6_ seems to be special and either does not belong to the L1 family or does not have size "c". WTF?!

```
 x   stm32l100rct6          24.0s
In file included from modm/src/modm/platform/dma/dma.cpp:12:
modm/src/modm/platform/dma/dma.hpp: In instantiation of 'class modm::platform::DmaController<2>::Channel<modm::platform::DmaBase::Channel::Channel3>':
modm/src/modm/platform/dma/dma.cpp:69:43:   required from here
modm/src/modm/platform/dma/dma.hpp:75:5: error: static assertion failed: invalid Channel ID for that DMA controller
   72 |     (ID == 1 and
      |     ~~~~~~~~~~~~
   73 |       ChannelID >= DmaBase::Channel::Channel1 and
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   74 |       ChannelID <= DmaBase::Channel::Channel7)
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   75 |     or
      |     ^~
   76 |     (ID == 2 and
      |     ~~~~~~~~~~~~
   77 |       ChannelID >= DmaBase::Channel::Channel1 and
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   78 |       ChannelID <= DmaBase::Channel::Channel2)
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
modm/src/modm/platform/dma/dma.hpp: In instantiation of 'class modm::platform::DmaController<2>::Channel<modm::platform::DmaBase::Channel::Channel4>':
modm/src/modm/platform/dma/dma.cpp:74:43:   required from here
modm/src/modm/platform/dma/dma.hpp:75:5: error: static assertion failed: invalid Channel ID for that DMA controller
modm/src/modm/platform/dma/dma.hpp: In instantiation of 'class modm::platform::DmaController<2>::Channel<modm::platform::DmaBase::Channel::Channel5>':
modm/src/modm/platform/dma/dma.cpp:79:43:   required from here
modm/src/modm/platform/dma/dma.hpp:75:5: error: static assertion failed: invalid Channel ID for that DMA controller
scons: *** [build/scons-release/modm/src/modm/platform/dma/dma.o] Error 1
```

Reply:
I think I solved it. There was a problem, with operator priorities of `and` and `or`. The F1 clause matched L100xC.
https://github.com/chris-durand/modm/commit/d9006824bc14f11882932a4c364370cf118191e5

Reply:
Thank you a lot!

Reply:
Yeah, I didn't spot that either, I looked into modm-devices data but gave up annoyed…